vm-monitor/src/main/java/com/epam/pipeline/vmmonitor/service/vm/VMMonitor.java

```diff
@@ -124,6 +128,23 @@ private boolean matchingRunExists(final VirtualMachine vm) {
         return false;
     }
 
+    private boolean poolIdExists(final VirtualMachine vm) {
+        log.debug("Checking whether a node pool with corresponding pool id exists.");
```

Comment:
As I mentioned previously `VirtualMachine ` doesn't have `pool` label, we need to check  `NodeInstance ` tags instead.